rclcpp/include/rclcpp/experimental/intra_process_manager.hpp

```diff
@@ -324,8 +301,39 @@ class IntraProcessManager
 
   struct SplittedSubscriptions
   {
-    std::vector<uint64_t> take_shared_subscriptions;
-    std::vector<uint64_t> take_ownership_subscriptions;
+    enum
+    {
+      IndexSharedTyped = 0, IndexSharedSerialized = 1,
+      IndexOwnershipTyped = 2, IndexOwnershipSerialized = 3,
+      IndexNum = 4
+    };
+
+    /// get the index for "take_subscriptions" depending on shared/serialized
+    constexpr static auto
+    get_index(const bool is_shared, const bool is_serialized)
+    {
+      return (is_serialized ? IndexSharedTyped : IndexSharedSerialized) +
+             (is_shared ? IndexSharedTyped : IndexOwnershipTyped);
```

Comment:
please, avoid the sum.
It might look like a clever way to calculate the index, but it's extremely confusing.

Reply:
It also looks like the first line was wrong:
```suggestion
      return (is_serialized ? IndexSharedSerialized : IndexSharedTyped) +
             (is_shared ? IndexSharedTyped : IndexOwnershipTyped);
```